irispy/sji.py

```diff
@@ -0,0 +1,181 @@
+
+
+def save_to_mp4(mc, outputfile, fps=60):
+    '''
+    Set up ffmpeg writer
+    Configures plot axes
+    Save an animation of the MapCube
+
+    Parameters
+    ----------
+    mc: `sunpy.map.MapCube`
+        Input mapcube
+    Outputfile: `str`
+        Path name of output file (.mp4)
+    fps: `int`
+        Desired frames per seconds (default=60)
+
+    Return
+    -----------
+    NoneType
+        mp4 to outputfile path
+    '''
+
+    plt.rcParams['animation.ffmpeg_path'] = '/usr/local/bin/ffmpeg'
+    writer = animation.FFMpegWriter(fps=fps, metadata=dict(artist='SunPy'),
+                                    bitrate=64000)
+    if not writer.isAvailable():
+        path = input('Set FFMpeg path: (ex. /usr/local/bin/ffmpeg)')
+        plt.rcParams['animation.ffmpeg_path'] = path
+    writer = animation.FFMpegWriter(fps=fps, metadata=dict(artist='SunPy'),
+                                    bitrate=64000)
+    if not writer.isAvailable():
+        print('Cannot find available FFMpeg Writer')
+        return
+    plt.rcParams['axes.facecolor'] = 'black'
+    plt.rcParams['xtick.direction'] = 'out'
+    plt.rcParams['ytick.direction'] = 'out'
+    ax = plt.axes(zorder=0)
+
+    ani = mc.plot(ax)
+    ani.save(outputfile, writer=writer)
+    print('Successfully saved to ' + outputfile)
+
+
+def dustbuster(filelist, clobber=False):
+    """
+       Read SJI fits files and return Inpaint-corrected fits files.
+       Image inpainting involves filling in part of an image or video
+       using information from the surrounding area.
+
+       **Warning** Will clobber input file.
+
+       Parameters
+       ----------
+       filelist: `string` or `list`
+           File to read, if single file string detected, will
+           create a list of length 1.
+
+
+       Returns
+       -------
+       NoneType
+           Inpaint-corrected fits to clobber input file
+       """
+
+    if type(filelist) == str:  # convert single file to list
+        filelist = [filelist]
+    nfile = 0
+    for file in filelist:
+        nfile += 1
+        image_header = fits.getheader(file)
+        image_orig = fits.getdata(file)
+        nx = image_header.get('NRASTERP')
+        # TODO mapcube input
+
+        img_shape = image_orig.shape
+        ndx = img_shape[0]
+
+        image_result = []
+        firstpos = range(ndx)[0::nx]
+
+        for i in range(0, ndx):
+            #  Create mask with values < 1, excluding frame (-200)
+            mask = np.zeros(image_orig[i].shape)
+            mask[np.where(image_orig[i] < 1)] = 1
+            mask[np.where(image_orig[i] == -200)] = 0
+
+            image_fix = image_orig[i].copy()
+            if nx <= 50:     # sparse/coarse raster
+                skip = 1
+            elif nx > 50:    # dense raster
+                skip = 3
+                secpos = range(ndx)[1::nx]
+                thirdpos = range(ndx)[2::nx]
+
+            if (i in firstpos) or (i in secpos) or (i in thirdpos):
```

Comment:
I tried running this function on the following file ```iris_l2_20141211_191222_3803105278_SJI_1400_t000.fits``` and got the following error at line 162: ```UnboundLocalError: local variable 'secpos' referenced before assignment```.